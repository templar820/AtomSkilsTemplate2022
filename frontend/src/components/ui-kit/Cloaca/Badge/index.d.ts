import * as React from 'react';
import { EmotionStylesType } from 'lib-ui';

declare type PositionX = 'right' | 'middle' | 'left';
declare type PositionY = 'top' | 'middle' | 'bottom';

export interface BadgeProps {
    /** Color for badge text */
    color?: string;
    /** Color for badge background */
    bgColor?: string;
    /** Defines whether badge should be outlined */
    isOutlined?: boolean;
    /** Defines badge's horisontal position */
    badgePositionX?: PositionX;
    /** Defines badge's horisontal position */
    x?: PositionX;
    /** Defines badge's vertical position */
    badgePositionY?: PositionY;
    /** Defines badge's vertical position */
    y?: PositionY;
    /** Defines whether any node will be placed in the body of badge */
    children?: React.ReactNode;
    /** Style group: badge label */
    extendStyle?: EmotionStylesType;
    /** Style group: outer element */
    outerStyle?: EmotionStylesType;
    /** Defines allignment of inner items */
    alignItems?: string;
    /** Defines badge icon */
    icon?: React.ReactNode;
}

declare const Badge: React.ComponentType<BadgeProps>;

export default Badge;
