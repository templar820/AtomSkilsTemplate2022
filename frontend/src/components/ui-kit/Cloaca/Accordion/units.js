import { css } from '@emotion/core';
import styled from '@emotion/styled';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const StyledPanelInner = styled.div`
    margin-bottom: 40px;
    padding-bottom: 1px;
    box-sizing: border-box;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledPanelHead = styled.div`
    ${({ colors, expanded, disabled }) => css`
        position: relative;
        min-height: 72px;
        font-weight: 500;
        font-size: 15px;
        line-height: 1;
        color: ${colors.GrayScale_700};
        border-top: 1px solid ${colors.GrayScale_200};
        border-bottom: 1px solid ${colors.GrayScale_200};
        cursor: pointer;
        padding: 30px 24px 1px 24px;
        background-color: ${colors[expanded ? 'GrayScale_100' : 'GrayScale_0']};
        ${disabled &&
            css`
                opacity: 0.3;
                cursor: not-allowed;
            `};

        &:after {
            content: '';
            position: absolute;
            right: 20px;
            top: 30px;
            width: 9px;
            height: 9px;
            border-right: 2px solid ${colors.GrayScale_700};
            border-bottom: 2px solid ${colors.GrayScale_700};
            transform: rotate(${expanded ? '-135deg' : '45deg'});
            transition: 0.5s transform ease;
        }
    `};
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const collapseContent = (contentHeight, expanded, animated) => {
    if (expanded === true) {
        return css`
            ${animated ? 'max-height: none;' : ' max-height: ' + contentHeight + 'px;'}
        `;
    } else {
        return css`
            ${animated ? 'max-height: ' + contentHeight + 'px;' : ' max-height: 0px;'}
        `;
    }
};

const StyledContentCollapse = styled.div`
    overflow: hidden;
    transition: 0.3s max-height ease;

    ${({ expanded, contentHeight, animated }) => collapseContent(contentHeight, expanded, animated)};
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledPanelContent = styled.div`
    overflow: hidden;
    padding: 93px 72px 10px 40px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const LoaderComponentWrapper = styled.div`
    position: relative;
    height: 50px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const EmptyStatesWrapper = styled.div`
    position: relative;
    min-height: 200px;
    ${({ styles, ...restProps }) => applyEmotionStyle(styles, restProps)};
`;

const StyledAccordionWrapper = styled.div`
    ${({ wrapperStyle, ...restProps }) => applyEmotionStyle(wrapperStyle, restProps)};
`;

export {
    StyledPanelInner,
    StyledPanelHead,
    StyledPanelContent,
    StyledContentCollapse,
    LoaderComponentWrapper,
    EmptyStatesWrapper,
    StyledAccordionWrapper
};
