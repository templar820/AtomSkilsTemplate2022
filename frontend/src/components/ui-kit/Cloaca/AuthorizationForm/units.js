import { css } from '@emotion/core';
import styled from '@emotion/styled';
import Icon from 'lib-ui/InlineIcons';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const Header = styled.h2`
    ${({ headerStyle, ...restProps }) => applyEmotionStyle(headerStyle, restProps)};
`;

const IsBlockedWrapper = styled.div`
    ${({ isBlockedStyle, ...restProps }) => applyEmotionStyle(isBlockedStyle, restProps)};
`;

const fieldStyle = (additionalStyle = null) => css`
    max-width: 100%;
    ${additionalStyle};
`;

const buttonStyle = (additionalStyle = null) => css`
    width: 100%;
    ${additionalStyle};
`;

const StyledIcon = styled(Icon)`
    cursor: pointer;
`;

export { Header, IsBlockedWrapper, fieldStyle, buttonStyle, StyledIcon };
