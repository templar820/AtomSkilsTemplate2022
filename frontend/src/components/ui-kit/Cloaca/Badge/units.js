/** @jsx jsx */
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const PosX = {
    right: 'right: 0; margin-left: 6px;',
    middle: 'left: calc(100% - 18px);',
    left: 'left: 0; order: -1; margin-left: 0; margin-right: 6px;'
};

const PosY = {
    top: 'top: 0;',
    middle: 'position: relative; left: initial;',
    bottom: 'bottom: 0;'
};

const StyledOuter = styled.div`
    display: flex;
    position: relative;
    margin-bottom: 10px;
    ${({ outerStyle, ...restProps }) => applyEmotionStyle(outerStyle, restProps)};
`;

const StyledInner = styled.div`
    position: relative;
    display: flex;
    align-items: ${({ alignItems }) => alignItems};
`;

const outlinedStyle = (colors) => css`
    color: ${colors.GrayScale_700};
    background: ${colors.GrayScale_0};
    border: 1px solid ${colors.GrayScale_700};
`;

const StyledBadgeLabel = styled.div`
    ${({
        colors,
        color,
        bgColor,
        badgePositionX,
        x,
        badgePositionY,
        y,
        icon,
        isOutlined,
        extendStyle,
        text,
        ...restProps
    }) => css`
        position: absolute;
        box-sizing: border-box;
        font-weight: 600;
        font-size: 10px;
        line-height: 1;
        text-align: center;
        border-radius: 16px;
        padding: 3px;
        color: ${color}; 
        background: ${bgColor || 'none'}; 
    
        ${PosX[x || badgePositionX]}
        ${PosY[y || badgePositionY]}
        max-height:${text ? '16px' : '6px'};
        ${icon && 'padding: 0px;  max-height: 16px;'}
        ${isOutlined && outlinedStyle(colors)}
        ${applyEmotionStyle(extendStyle, {
            colors,
            color,
            bgColor,
            badgePositionX,
            x,
            badgePositionY,
            y,
            icon,
            isOutlined,
            text,
            ...restProps
        })};
    `}
`;

export { StyledInner, StyledBadgeLabel, StyledOuter };
