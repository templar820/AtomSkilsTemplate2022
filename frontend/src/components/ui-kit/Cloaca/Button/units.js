import React from 'react';

import { css, keyframes } from '@emotion/core';
import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';

import { hexToRGBA, pickColor } from 'lib-root/utils/styleMixins';

import Icon from 'lib-ui/InlineIcons';

import config from './config';

import { btnSizes, _lighten } from './utils';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const fadeInAnimation = keyframes`
          100% {
            opacity: 1;
          }
        `;

const baseStyles = ({ icon, size = 'lg', fontSize, color: customColor, colorDiff, isLoading, colors }) => css`
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: ${icon ? '24px' : 'auto'};
    width: ${icon ? btnSizes(fontSize)[size].iconBtnWidth : 'auto'};
    height: ${btnSizes(fontSize)[size].height};
    margin: 0;
    padding: ${icon ? '0' : `5px ` + btnSizes(fontSize)[size].vPadding};
    border: 1px solid;
    border-radius: ${icon ? '100%' : '100px'};
    font-weight: 500;
    font-size: ${fontSize ? (typeof fontSize === 'number' ? fontSize + 'px' : fontSize) : '15px'};
    line-height: 1.6;
    cursor: pointer;
    user-select: none;
    transition: 0.3s all;
    color: ${colors.GrayScale_0};
    white-space: nowrap;
    ${isLoading && 'pointer-events: none;'}
    &.hover,
    &:hover {
        background-color: ${pickColor(colors.primaryAccent, _lighten, customColor, { colorDiff })};
        border-color: ${pickColor(colors.primaryAccent, _lighten, customColor, { colorDiff })};
        color: ${colors.GrayScale_0};
    }
    &:focus {
        outline: 0;
    }
    &:active,
    &.active {
        border-color: transparent;
        background-color: ${colors.tertiaryAccent};
        box-shadow: inset 0 1px 3px ${colors.GrayScale_700};
        color: ${colors.GrayScale_0};
    }
    &:disabled {
        background-color: ${colors.GrayScale_200};
        border-color: ${colors.GrayScale_200};
        cursor: not-allowed;
        color: ${colors.GrayScale_0};
    }
    &:after {
        content: '';
        min-height: inherit;
        font-size: 0;
    }
`;

const colorSchemeStyles = {
    secondary: ({ color: customColor, colorDiff, colors }) => css`
        &:not(:disabled) {
            &:not(:active):not(.active) {
                background-color: ${pickColor(colors.secondary, null, customColor)};
                border-color: ${pickColor(colors.secondary, null, customColor)};
                &:hover,
                &.hover {
                    background-color: ${pickColor(colors.secondaryAccent, _lighten, customColor, {
                        colorDiff
                    })};
                    border-color: ${pickColor(colors.secondaryAccent, _lighten, customColor, {
                        colorDiff
                    })};
                }
            }
        }
    `,
    white: ({ colors }) => css`
        background-color: ${colors.GrayScale_0};
        border-color: ${colors.GrayScale_0};
        color: ${colors.GrayScale_700};
        &:active,
        &.active {
            background-color: transparent;
            border: 1px solid ${colors.GrayScale_0};
            box-shadow: none;
        }
        &:disabled {
            background-color: ${colors.GrayScale_100};
            color: ${hexToRGBA(colors.GrayScale_700, 0.5)};
        }
    `,
    outline: ({ color: customColor, colors }) => css`
        background-color: transparent;
        border-color: ${pickColor(colors.GrayScale_0, null, customColor)};
        color: ${pickColor(colors.GrayScale_0, null, customColor)};
        &:active,
        &.active,
        &:disabled {
            background-color: transparent;
            border: 1px solid ${pickColor(colors.GrayScale_0, null, customColor)};
            color: ${pickColor(colors.GrayScale_0, null, customColor)};
            box-shadow: none;
        }
        &:disabled {
            opacity: 0.5;
        }
        &:not(:disabled) {
            &:not(:active):not(.active) {
                &.hover,
                &:hover {
                    background-color: ${pickColor(colors.primaryAccent, null, customColor)};
                    border-color: ${pickColor(colors.primaryAccent, null, customColor)};
                }
            }
        }
    `,
    primary: ({ color: customColor, colors }) => css`
        background-color: ${pickColor(colors.primary, null, customColor)};
        border-color: ${pickColor(colors.primary, null, customColor)};
    `
};

const StyledButton = styled('button', {
    shouldForwardProp: (prop) => isPropValid(prop) && prop !== 'isLoading'
})`
    ${baseStyles}
    ${(props) => (colorSchemeStyles[props.colorScheme || 'primary'] || colorSchemeStyles.primary)(props)}
    ${({ css, ...restProps }) => applyEmotionStyle(css, restProps)};
`;

const StyledRightIcon = styled(Icon)`
    ${({ isRotate }) => (isRotate ? 'transform: rotate(270deg);' : '')}
`;

const StyledLeftIcon = styled(Icon)`
    margin-right: 4px;
`;

const StyledContentWrapper = styled.span`
    margin-right: ${({ rightIcon }) => (rightIcon ? '-16px' : '0')};
    min-width: 60px;
    opacity: 0;
    animation: ${fadeInAnimation} ${config.animation.duration}ms forwards;
`;

const StyledButtonInner = styled('div', {
    shouldForwardProp: (prop) => isPropValid(prop) && prop !== 'isLoading'
})`
    display: flex;
    justify-content: center;
`;

const Fade = styled.div`
    opacity: 0;
    animation: ${fadeInAnimation} ${config.animation.duration}ms forwards;
`;

export {
    baseStyles,
    colorSchemeStyles,
    StyledButton,
    StyledRightIcon,
    StyledLeftIcon,
    StyledContentWrapper,
    StyledButtonInner,
    Fade
};
