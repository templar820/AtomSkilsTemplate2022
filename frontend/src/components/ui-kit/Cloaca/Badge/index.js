import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { ColorsContext } from 'lib-root/colors';
import Icon from 'lib-ui/InlineIcons';
import { StyledInner, StyledBadgeLabel, StyledOuter } from './units';
import { setClassName } from 'lib-root/utils/styleMixins';

/**
 * `Badge` используется для выделения ключевой информации с предустановленным статусом
 */
const Badge = React.forwardRef(
    ({ className, color, bgColor, icon, text, badgePositionX, children, outerStyle, alignItems, ...rest }, ref) => {
        const colors = useContext(ColorsContext);
        bgColor = colors[bgColor] || bgColor;
        color = colors[color] || color;

        const childrenCount = React.Children.count(children);

        return (
            <StyledOuter {...{ ref, outerStyle }} className={setClassName({ props: { className }, name: 'badge' })}>
                <StyledInner {...{ badgePositionX, alignItems }}>
                    {children}
                    <StyledBadgeLabel
                        {...{
                            colors,
                            color,
                            bgColor,
                            text,
                            icon,
                            badgePositionX,
                            childrenCount
                        }}
                        {...rest}>
                        {icon && (
                            <Icon
                                color={color || colors.errorAccent}
                                width={bgColor ? '16px' : '18px'}
                                height={bgColor ? '16px' : '18px'}
                                icon={icon}
                            />
                        )}
                        {text}
                    </StyledBadgeLabel>
                </StyledInner>
            </StyledOuter>
        );
    }
);

Badge.displayName = 'Badge';
Badge.propTypes = {
    /** Color for badge text */
    color: PropTypes.string,
    /** Color for badge background */
    bgColor: PropTypes.string,
    /** Defines whether badge should be outlined */
    isOutlined: PropTypes.bool,
    /** Defines badge's horisontal position */
    badgePositionX: PropTypes.oneOf(['right', 'middle', 'left']),
    /** Defines badge's horisontal position */
    x: PropTypes.oneOf(['right', 'middle', 'left']),
    /** Defines badge's vertical position */
    badgePositionY: PropTypes.oneOf(['top', 'middle', 'bottom']),
    /** Defines badge's vertical position */
    y: PropTypes.oneOf(['top', 'middle', 'bottom']),
    /** Defines whether any node will be placed in the body of badge */
    children: PropTypes.node,
    /** Style group: badge label */
    extendStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Style group: outer element */
    outerStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Defines allignment of inner items */
    alignItems: PropTypes.string,
    /** Defines badge icon */
    icon: PropTypes.oneOfType([PropTypes.node, PropTypes.string])
};

Badge.defaultProps = {
    color: 'GrayScale_0',
    bgColor: 'errorAccent',
    badgePositionX: 'right',
    badgePositionY: 'top',
    alignItems: 'center'
};

export default Badge;
