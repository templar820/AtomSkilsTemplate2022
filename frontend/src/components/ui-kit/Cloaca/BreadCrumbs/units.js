import styled from '@emotion/styled';

import { rem } from 'lib-root/utils/units';
import { hexToRGBA } from 'lib-root/utils/styleMixins';
import Icon from 'lib-ui/InlineIcons';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const Wrapper = styled.div`
    position: relative;
    flex: 0 0 ${rem(22)};
    margin-right: ${rem(8)};
    width: ${({ width }) => width};
    height: ${({ height }) => height};
    cursor: default;
    font-size: ${({ size }) => (size === 'sm' ? '16px' : '20px')};
    font-weight: 500;
    ${({ wrapStyle, ...restProps }) => applyEmotionStyle(wrapStyle, restProps)};
`;

const StyledLink = styled.a`
    margin-right: 5px;
    color: ${({ disabled, colors }) => (disabled ? `${hexToRGBA(colors.GrayScale_700, 0.5)}  !important` : 'inherit')};
    pointer-events: ${({ disabled }) => (disabled ? 'none !important' : 'auto')};
    user-select: ${({ disabled }) => (disabled ? 'none !important' : 'auto')};
    cursor: pointer;
    &:hover {
        text-decoration: none;
        color: inherit;
    }
`;
const StyledText = styled.span`
    margin-right: 5px;
    color: ${({ lastItemColor }) => lastItemColor};
    ${({ textStyles, ...restProps }) => applyEmotionStyle(textStyles, restProps)};
`;

const StyledItem = styled.span`
    margin-right: 5px;
    display: inline-flex;
    align-items: center;
    color: inherit;
    ${({ itemStyle, ...restProps }) => applyEmotionStyle(itemStyle, restProps)};
`;

const StyledIcon = styled(Icon)`
    margin-right: 5px;
    transform: rotate(270deg);
    ${({ iconStyle, ...restProps }) => applyEmotionStyle(iconStyle, restProps)};
`;

export { StyledIcon, StyledItem, StyledLink, StyledText, Wrapper };
