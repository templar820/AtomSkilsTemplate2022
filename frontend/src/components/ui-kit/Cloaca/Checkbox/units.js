import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { autoAddPx } from 'lib-root/utils/styleMixins';
import { px } from 'lib-root/utils/units';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const StyledLabel = styled.label`
    display: inline-flex;
    align-items: center;
    margin: 0;
    ${({ isChildrenExist, size }) => {
        return (
            isChildrenExist &&
            css`
                padding-right: ${px(size, 0.375)};
            `
        );
    }};
    font-size: ${({ size }) => autoAddPx(size)};
    cursor: pointer;
    user-select: none;
    ${({ disabled }) => {
        return (
            disabled &&
            css`
                opacity: 0.25;
                cursor: not-allowed;
            `
        );
    }}

    ${({ _css, ...restProps }) => applyEmotionStyle(_css, restProps)};
`;

const CheckboxSquare = styled.span`
    ${({ colors, isChildrenExist, disabled, isHovered, _checked, squareStyle, size, ...restProps }) => css`
        position: relative;
        display: inline-block;
        width: 20px;
        min-width: 20px;
        height: 20px;
        min-height: 20px;
        margin: ${px(size, 0.375)};
        ${isChildrenExist &&
            css`
                margin-right: ${px(size, 0.875)};
            `};
        border-radius: ${px(size, 0.125)};
        background-color: ${disabled ? colors.GrayScale_200 : 'transparent'};
        color: ${isHovered ? colors.primaryAccent : colors.primary};
        border: ${px(size, 0.125)} solid
            ${disabled ? colors.GrayScale_200 : isHovered ? 'currentColor' : colors.GrayScale_200};
        .hover & {
            border-color: ${colors.primaryAccent};
            color: ${colors.primaryAccent};
        }
        svg {
            width: ${px(size, 1.25)};
            height: ${px(size, 1.25)};
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
        }
        ${_checked === 'indeterminate' &&
            css`
                &:after {
                    content: '';
                    position: absolute;
                    top: 50%;
                    left: 50%;
                    transform: translate(-50%, -50%);
                    width: ${px(size, 0.5)};
                    height: ${px(size, 0.5)};
                    border-radius: ${px(size, 0.0625)};
                    background-color: ${disabled ? colors.primary : 'currentColor'};
                }
            `};

        ${applyEmotionStyle(squareStyle, {
            colors,
            isChildrenExist,
            disabled,
            isHovered,
            _checked,
            size,
            ...restProps
        })};
    `}
`;

const CheckboxCaption = styled.span`
    overflow: hidden;
    font-size: 15px;
    line-height: 24px;
    ${({ captionStyle }) => captionStyle}
`;

const CheckboxSvg = ({ colors, disabled, _checked }) =>
    _checked === true && (
        <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path
                d="M8.83178 11.5475L6.61877 9.33452L5.20483 10.7485L7.66627 13.2102L9.11181 14.656L15.044 6.16961L13.4057 5.02246L8.83178 11.5475Z"
                fill={disabled ? colors.primary : 'currentColor'}
            />
        </svg>
    );

export { StyledLabel, CheckboxSquare, CheckboxCaption, CheckboxSvg };
