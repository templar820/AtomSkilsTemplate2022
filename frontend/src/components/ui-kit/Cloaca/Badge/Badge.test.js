import React from 'react';
import ReactDOM from 'react-dom';
import serializer from 'jest-emotion';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Badge from './';
import { StyledInner, StyledBadgeLabel, StyledOuter } from './units';

expect.addSnapshotSerializer(serializer);

Enzyme.configure({ adapter: new Adapter() });

const props = {
    children: <div id="child">children</div>,
    badgePositionX: 'left',
    color: 'textCollor',
    bgColor: 'badgeColor',
    text: '99+',
    icon: 'Information',
    outerStyle: { marginLeft: '50px' },
    alignItems: 'right',
    isOutlined: true,
    x: 'right',
    badgePositionY: 'middle',
    y: 'middle',
    extendStyle: { marginLeft: '50px' }
};

describe('<Badge />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Badge {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Badge />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        const label = wrapper.find(StyledBadgeLabel);
        expect(wrapper.contains(props.children)).toBeTruthy();
        expect(wrapper.find('#child').length).toBe(1);
        expect(wrapper.props()).toHaveProperty('badgePositionX', props.badgePositionX);
        expect(label.prop('badgePositionX')).toEqual(props.badgePositionX);
        expect(wrapper.props()).toHaveProperty('color', props.color);
        expect(label.prop('color')).toEqual(props.color);
        expect(wrapper.props()).toHaveProperty('bgColor', props.bgColor);
        expect(label.prop('bgColor')).toEqual(props.bgColor);
        expect(wrapper.props()).toHaveProperty('text', props.text);
        expect(label.prop('text')).toEqual(props.text);
        expect(wrapper.props()).toHaveProperty('icon', props.icon);
        expect(label.prop('icon')).toEqual(props.icon);
        expect(wrapper.props()).toHaveProperty('outerStyle', props.outerStyle);
        expect(wrapper.find(StyledOuter).prop('outerStyle')).toEqual(props.outerStyle);
        expect(wrapper.props()).toHaveProperty('alignItems', props.alignItems);
        expect(wrapper.find(StyledInner).prop('alignItems')).toEqual(props.alignItems);
        expect(wrapper.children()).toHaveLength(React.Children.count(props.children));
        expect(wrapper.props()).toHaveProperty('isOutlined', props.isOutlined);
        expect(label.prop('isOutlined')).toEqual(props.isOutlined);
        expect(wrapper.props()).toHaveProperty('x', props.x);
        expect(label.prop('x')).toEqual(props.x);
        expect(wrapper.props()).toHaveProperty('badgePositionY', props.badgePositionY);
        expect(label.prop('badgePositionY')).toEqual(props.badgePositionY);
        expect(wrapper.props()).toHaveProperty('y', props.y);
        expect(label.prop('y')).toEqual(props.y);
        expect(wrapper.props()).toHaveProperty('extendStyle', props.extendStyle);
        expect(label.prop('extendStyle')).toEqual(props.extendStyle);
    });
});
