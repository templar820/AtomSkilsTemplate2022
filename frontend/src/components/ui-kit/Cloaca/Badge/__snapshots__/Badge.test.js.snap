// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should match snapshot 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-bottom: 10px;
  margin-left: 50px;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: right;
  -webkit-box-align: right;
  -ms-flex-align: right;
  align-items: right;
}

.emotion-2 {
  position: absolute;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 10px;
  line-height: 1;
  text-align: center;
  border-radius: 16px;
  padding: 3px;
  color: textCollor;
  background: badgeColor;
  right: 0;
  margin-left: 6px;
  position: relative;
  left: initial;
  max-height: 16px;
  padding: 0px;
  max-height: 16px;
  color: #333333;
  background: #ffffff;
  border: 1px solid #333333;
  margin-left: 50px;
}

.emotion-2 {
  position: absolute;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 10px;
  line-height: 1;
  text-align: center;
  border-radius: 16px;
  padding: 3px;
  color: textCollor;
  background: badgeColor;
  right: 0;
  margin-left: 6px;
  position: relative;
  left: initial;
  max-height: 16px;
  padding: 0px;
  max-height: 16px;
  color: #333333;
  background: #ffffff;
  border: 1px solid #333333;
  margin-left: 50px;
}

.emotion-0 {
  fill: textCollor;
  width: 16px;
  min-width: 16px;
  height: 16px;
}

.emotion-0 {
  fill: textCollor;
  width: 16px;
  min-width: 16px;
  height: 16px;
}

<Badge
  alignItems="right"
  badgePositionX="left"
  badgePositionY="middle"
  bgColor="badgeColor"
  color="textCollor"
  extendStyle={
    Object {
      "marginLeft": "50px",
    }
  }
  icon="Information"
  isOutlined={true}
  outerStyle={
    Object {
      "marginLeft": "50px",
    }
  }
  text="99+"
  x="right"
  y="middle"
>
  <StyledOuter
    className="CLR-badge"
    outerStyle={
      Object {
        "marginLeft": "50px",
      }
    }
  >
    <div
      className="CLR-badge emotion-6 emotion-7"
    >
      <StyledInner
        alignItems="right"
        badgePositionX="left"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <div
            id="child"
          >
            children
          </div>
          <StyledBadgeLabel
            badgePositionX="left"
            badgePositionY="middle"
            bgColor="badgeColor"
            childrenCount={1}
            color="textCollor"
            colors={
              Object {
                "GrayScale_0": "#ffffff",
                "GrayScale_100": "#EDEDED",
                "GrayScale_150": "#E5E5E5",
                "GrayScale_200": "#D8D8D8",
                "GrayScale_50": "#fafafa",
                "GrayScale_500": "#999999",
                "GrayScale_55": "#f6f6f6",
                "GrayScale_600": "#7C7C7C",
                "GrayScale_700": "#333333",
                "GrayScale_900": "#000000",
                "error": "#F62434",
                "errorAccent": "#EB5A40",
                "info": "#34AAF2",
                "infoAccent": "#5BD9E5",
                "primary": "#00B956",
                "primaryAccent": "#10E272",
                "secondary": "#731982",
                "secondaryAccent": "#A500BF",
                "tertiary": "#444189",
                "tertiaryAccent": "#404D46",
                "warning": "#FFA717",
                "warningAccent": "#F36F20",
              }
            }
            extendStyle={
              Object {
                "marginLeft": "50px",
              }
            }
            icon="Information"
            isOutlined={true}
            text="99+"
            x="right"
            y="middle"
          >
            <div
              className="emotion-2 emotion-3"
              color="textCollor"
              x="right"
              y="middle"
            >
              <withColors(InlineIcons)
                color="textCollor"
                disabled={false}
                h={null}
                height="16px"
                icon="Information"
                w={null}
                width="16px"
              >
                <InlineIcons
                  color="textCollor"
                  colors={
                    Object {
                      "GrayScale_0": "#ffffff",
                      "GrayScale_100": "#EDEDED",
                      "GrayScale_150": "#E5E5E5",
                      "GrayScale_200": "#D8D8D8",
                      "GrayScale_50": "#fafafa",
                      "GrayScale_500": "#999999",
                      "GrayScale_55": "#f6f6f6",
                      "GrayScale_600": "#7C7C7C",
                      "GrayScale_700": "#333333",
                      "GrayScale_900": "#000000",
                      "error": "#F62434",
                      "errorAccent": "#EB5A40",
                      "info": "#34AAF2",
                      "infoAccent": "#5BD9E5",
                      "primary": "#00B956",
                      "primaryAccent": "#10E272",
                      "secondary": "#731982",
                      "secondaryAccent": "#A500BF",
                      "tertiary": "#444189",
                      "tertiaryAccent": "#404D46",
                      "warning": "#FFA717",
                      "warningAccent": "#F36F20",
                    }
                  }
                  disabled={false}
                  h={null}
                  height="16px"
                  icon="Information"
                  w={null}
                  width="16px"
                >
                  <Styled(svg)
                    className="CLR-inline-icons"
                    color="textCollor"
                    colors={
                      Object {
                        "GrayScale_0": "#ffffff",
                        "GrayScale_100": "#EDEDED",
                        "GrayScale_150": "#E5E5E5",
                        "GrayScale_200": "#D8D8D8",
                        "GrayScale_50": "#fafafa",
                        "GrayScale_500": "#999999",
                        "GrayScale_55": "#f6f6f6",
                        "GrayScale_600": "#7C7C7C",
                        "GrayScale_700": "#333333",
                        "GrayScale_900": "#000000",
                        "error": "#F62434",
                        "errorAccent": "#EB5A40",
                        "info": "#34AAF2",
                        "infoAccent": "#5BD9E5",
                        "primary": "#00B956",
                        "primaryAccent": "#10E272",
                        "secondary": "#731982",
                        "secondaryAccent": "#A500BF",
                        "tertiary": "#444189",
                        "tertiaryAccent": "#404D46",
                        "warning": "#FFA717",
                        "warningAccent": "#F36F20",
                      }
                    }
                    disabled={false}
                    height="16px"
                    width="16px"
                  >
                    <svg
                      className="CLR-inline-icons emotion-0 emotion-1"
                      color="textCollor"
                      disabled={false}
                      height="16px"
                      width="16px"
                    >
                      <use
                        xlinkHref="#Information"
                      />
                    </svg>
                  </Styled(svg)>
                </InlineIcons>
              </withColors(InlineIcons)>
              99+
            </div>
          </StyledBadgeLabel>
        </div>
      </StyledInner>
    </div>
  </StyledOuter>
</Badge>
`;
